tinygrad/renderer/hip.py

```diff
@@ -22,7 +20,10 @@ class HIPLanguage(CStyleLanguage):
   uses_vload=True
   uses_ptr_arithmetic=True
   arg_int_prefix = "const int"
-  half_prekernel = "#include <hip/hip_fp16.h>\nusing half4 = HIP_vector_type<half, 4>;" + """
+  half_prekernel = "#include <hip/hip_fp16.h>\n" + """
+typedef union { struct { half x, y, z, w; }; half data[4]; } half4; __device__ half4 make_half4(half x, half y, half z, half w) { return {x, y, z, w}; }
```

Comment:
would it make sense to add `__align__(8)` here which I didn't see in the header file? so that when compiles it will use the vector load?

Reply:
I added alignment of 8 and 16 for half4 and half8